addons/sale_expense/models/hr_expense.py

```diff
@@ -37,7 +40,11 @@ def _onchange_sale_order_id(self):
         self.env.add_to_compute(self._fields['analytic_distribution'], to_reset)
 
     def _get_split_values(self):
```

Comment:
That isn't how the split flow is supposed to work

When splitting an expense
the expense values are sent to the wizard to define the first two splits values. You can then edit those as "normal expenses" and eventually create new splits (as long as the total stays the same)
When validating, the original expense is modified and the other new expenses are created.
There are no correct answers to the question of the sale_order_id value though, as some types of expenses would like to keep it for each split an other shouldn't. This could be improved but it's hard to define which one should keep it.

The only thing the compute should do is remove it if it does not make sense to have it (E.G. put a non-reinvoiceable product on a split)